### DISCUSSION TITLE
fetch data ONE time for useQuery

### DISCUSSION THREAD

**Discussion by IRediTOTO** [blitz-js/blitz#3019] [Q&A] [answered]
Hi, can you make one more option for `useQuery` hook? 
I would love have an option like this
```
const [data] = useQuery(
    getUserProfile,
    { id: 1 },
    {  turnOffRefetch: true}  // I just need fetch one time to push data to the form init 
  )
```

I usually use `SWR` they have this option like that [https://swr.vercel.app/docs/revalidation#disable-automatic-revalidations](https://swr.vercel.app/docs/revalidation#disable-automatic-revalidations)

**Comment by abuuzayr** [comment, marked as answer]
Hi @IRediTOTO , you should be able to use the `staleTime` option to disable refetching:

```
const [data] = useQuery(
    getUserProfile,
    { id: 1 },
    { staleTime: Infinity }  // The data will never be considered stale, i.e. no refetching is done
  )
```

The rest of the options are here: https://blitzjs.com/docs/use-query
